SCClassLibrary/Common/GUI/Base/QWebView.sc

```diff
@@ -1,49 +1,172 @@
+QCallback : QObject {
+	var <action;
+
+	*qtClass { ^'QtCollider::QcCallback'; }
+
+	*new {
+		|func|
+		^super.new.action_(func)
+	}
+
+	*newFrom {
+		|other|
+		if (other.isKindOf(QCallback)) {
+			^QCallback(other.action)
+		} {
+			^QCallback(other)
+		}
+	}
+
+	action_{
+		|newAction|
+		if (newAction != action) {
+			this.prOnCalledSignals.do({
+				|signal|
+				if (action.notNil) { this.disconnectFunction(signal.asSymbol, action) };
+				if (newAction.notNil) { this.connectFunction(signal.asSymbol, { |cb, v| newAction.(v) }, true) }
+			});
+			action = newAction;
+		}
+	}
+
+	value {
+		|...args|
+		action.value(*args)
+	}
+
+	prOnCalledSignals {
+		^this.methods(false, true, false).select({
+			|sig|
+			sig.asString.find("onCalled").notNil
+		})
+	}
+}
+
 WebView : View {
+	classvar urlHandlers;
 
-	var <onLoadFinished, <onLoadFailed, <onLinkActivated, <onReload, <onJavaScriptMsg;
+	var <onLoadFinished, <onLoadFailed, <onLoadProgress, <onLoadStarted, <onLinkActivated, <onLinkHovered, <onReloadTriggered, <onJavaScriptMsg,
+		<onSelectionChanged, <onTitleChanged, <onUrlChanged, <onScrollPositionChanged, <onContentsSizeChanged, <onAudioMutedChanged,
+		<onRecentlyAudibleChanged;
 
 	*qtClass { ^'QtCollider::WebView'; }
 
+	*initClass {
+		urlHandlers = IdentityDictionary();
+	}
+
+	*setUrlHandler {
+		|prefix, function|
+		function = function.as(QCallback);
+		this.prSetUrlHandler(prefix, function);
+		urlHandlers[prefix.asSymbol] = function;
+	}
+
+	*prSetUrlHandler {
+		|prefix, function|
+		_Qt_SetUrlHandler
+		^this.primitiveFailed
+	}
+
 	*clearCache {
 		_QWebView_ClearMemoryCaches
 		^this.primitiveFailed
 	}
 
-	url { ^this.getProperty( \url ); }
+	init {
+		this.connectMethod('interpret(QString)', 'onInterpret');
+		this.connectMethod('renderProcessTerminated(int,int)', 'onTerminated');
+	}
 
-	url_ { arg address; this.setProperty( \url, address ); }
+	setFontFamily {
+		|which, font|
+		this.invokeMethod('setFontFamily', [QWebFontFamily(which), font]);
+	}
 
-	title { ^this.getProperty( \title ); }
+	triggerPageAction {
+		|action, checked=true|
+		this.invokeMethod('triggerPageAction', [QWebPageAction(action), checked])
+	}
 
-	// Get the displayed content in html form.
-	html { ^this.getProperty( \html ); }
+	setHtml {
+		|html, baseUrl="sc:///"|
+		this.invokeMethod('setHtml', [html, baseUrl]);
+	}
 
-	// Set html content.
-	html_ { arg htmlString;
-		this.invokeMethod( \setHtml, htmlString );
+	toHtml {
+		|cb|
```

Comment:
in general SC interfaces use the argument name `action` for callbacks. given that parameter names are part of the public interface I think you should consider making it either `action` or `callback` for clarity, with a strong preference for `action`.